src/jmclient/configure.py

```diff
@@ -514,6 +514,11 @@ def jm_single() -> AttributeDict:
 # [fraction, 0-1] / variance around all offer sizes. Ex: 500k minsize, 0.1 var = 450k-550k
 size_factor = 0.1
 
+#
+minimum_tx_fee_rate = 1000
```

Comment:
Could you share some thinking behind this number? (I mean, apart from the obvious :) ).

Reply:
Nothing apart from the obvious - it's default minimum tx relay fee.

Reply:
Right. This might be related to point 3/ in my list of suggestions [here](https://github.com/JoinMarket-Org/joinmarket-clientserver/issues/721#issuecomment-1824627009).

You want this var to be continuously updated, at least as transactions are confirmed, but much more, also when new blocks come in, probably. I agree with you that we want that kind of frequency of update.

I didn't think about this in detail earlier today but ... YGs advertise their offers when they start, and also advertise updates in the pit on transaction confirmation or "arrival" (*sometimes*) but when the taker requests `!orderbook` they're going to privmsg their offer at that point. The problem is that that announcement-in-private (the response to the Taker's `!orderbook`) is handled entirely within the daemon, as per the comment to `src.jmdaemon.daemon_protocol.JMDaemonServerProtocol.on_orderbook_requested`, this is dealt with by the daemon assuming the offerlist is up to date.

If we wanted to make that a dynamic update, we'll have to add more code to inject updates to the offerlist, e.g. at a regular interval, or perhaps even, every time an orderbook request is received.

There's also a question of whether we want the starting value (default?) to be 1000, or maybe have a starting value be undefined, forcing it to be calculated or something like that.

Reply:
You mean that makers should never sign transactions that are below local `mempoolminfee` and thus will not be seen in local mempool, which means maker either will see it only after longer time when it is confirmed or will make full-RBF replacement without even knowing it?

If we cannot avoid dynamic offer updates after each block, then: 1) no point to not implement also block confimation target for this setting too, I avoided it to get MVP ready faster, 2) it is related to #1042, same problems holding it back needs to be solved.

Reply:
> If we cannot avoid dynamic offer updates after each block

I think we can, but, it might be the case that we need some extra code for client-daemon communication, whether we choose to update immediately when the taker sends us the `!orderbook` request, or, say, once per block. In either case, it is more dynamic than what we do now: we only update offer details when there is a trigger for `on_tx_unconfirmed` or `on_tx_confirmed`.

This "extra code" is an extra communication type specified in `jmbase.commands.py` and then extra functions to send/receive message in `jmdaemon.daemon_protocol.py` and `jmclient.client_protocol.py`. If we don't do that, then as mentioned above, the daemon just reads the current offer from an internally stored `offerlist` variable, and doesn't know that anything changed in `jmclient`.

Reply:
> You mean that makers should never sign transactions that are below local mempoolminfee and thus will not be seen in local mempool, which means maker either will see it only after longer time when it is confirmed or will make full-RBF replacement without even knowing it?

I didn't specifically mean that, but sure, that is a scenario where it could cause problems. I just don't understand why it would make sense to have the exact threshold of failure (1sat/vb) to propagate, as the default.

Reply:
Ok, another reason is that values below 1000 for tx feerates are treated as block confirmation targets not sats per kvB by `estimate_fee_per_kb()`.